src/fixture-factory.ts

```diff
@@ -0,0 +1,114 @@
+import {
+    FactoryBuildOptions,
+    FactoryFunction,
+    FactoryOptions,
+    FixtureStatic,
+} from './types';
+import { TypeFactory } from './type-factory';
+import {
+    fileAppendJson,
+    readFileIfExists,
+    saveFixture,
+    structuralMatch,
+} from './utils';
+
+type WriteBuildAsync<T> = (
+    path: string,
+    options?: FactoryBuildOptions<T>,
+) => Promise<T>;
+type WriteBuildSync<T> = (path: string, options?: FactoryBuildOptions<T>) => T;
+type WriteBatchAsync<T> = (
+    path: string,
+    size: number,
+    options?: FactoryBuildOptions<T>,
+) => Promise<T[]>;
+type WriteBatchSync<T> = (
+    path: string,
+    size: number,
+    options?: FactoryBuildOptions<T>,
+) => T[];
+
+type PropsBuildAsync = { batch: false; synchronous: false };
+type PropsBuildSync = { batch: false; synchronous: true };
+type PropsBatchAsync = { batch: true; synchronous: false };
+type PropsBatchSync = { batch: true; synchronous: true };
+
+const propsBuildAsync = { batch: false, synchronous: false };
+const propsBuildSync = { batch: false, synchronous: true };
+const propsBatchAsync = { batch: true, synchronous: false };
+const propsBatchSync = { batch: true, synchronous: true };
+
+export class FixtureFactory<T> extends TypeFactory<T> {
+    private defaultPath = '';
+
+    constructor(
+        defaults: FactoryOptions<T>,
+        factory?: FactoryFunction<T>,
+        defaultPath?: string,
+    ) {
+        super(defaults, factory);
+        if (defaultPath) this.defaultPath = defaultPath;
+    }
+
+    public static = {
+        build: this.writeMethod(propsBuildAsync as PropsBuildAsync),
+        buildSync: this.writeMethod(propsBuildSync as PropsBuildSync),
+        batch: this.writeMethod(propsBatchAsync as PropsBatchAsync),
+        batchSync: this.writeMethod(propsBatchSync as PropsBatchSync),
+    };
+
```

Comment:
So -

IMO the FixtureFactory class should extend TypeFactory but not modify the underlying methods it extends. What you are doing here is exposing the same method names but for the class - this is going to be very confusing.

I would completely remove this section.

Reply:
no, no!

there now is `fixtureFactory.build()` which is the same as `typeFactory.build`, but now there is also `fixtureFactory.static.build()`

Reply:
maybe we call it `factory.save.build()`, instead?
maybe we should make it a method, as well, for future-proof?
```ts
public save() {
    return {
        build: this.writeMethod(propsBuildAsync as PropsBuildAsync),
        buildSync: this.writeMethod(propsBuildSync as PropsBuildSync),
        batch: this.writeMethod(propsBatchAsync as PropsBatchAsync),
        batchSync: this.writeMethod(propsBatchSync as PropsBatchSync),
    }
}

factory.save().batch(5);
```